### PR TITLE
[MM-12476] Consistent paging arguments limit/offset vs page/perPage for plugin API

### LINKED ISSUE #9814: [Help Wanted] [MM-12476] Consistent paging arguments limit/offset vs page/perPage for plugin API
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-12476)

Some plugin API methods use `offset, limit`, while some use `page, perPage`. Let's converge on the latter for uniformity in the API.

This means changing the following API endpoints.
```go
	// GetTeamMembers returns the memberships of a specific team.
	GetTeamMembers(teamId string, offset, limit int) ([]*model.TeamMember, *model.AppError)

	// GetPublicChannelsForTeam gets a list of all channels.
	GetPublicChannelsForTeam(teamId string, offset, limit int) (*model.ChannelList, *model.AppError)
```

This ticket would be to propose a strategy to unify the API methods. Do we change them under the auspices of plugins being "in beta"? Do we introduce new versions and deprecate the old ones? Is there another strategy we could use?

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This changes the parameter of two plugin API methods to use `page, perPage` instead of `offset, limit`.
Both API methods just call the respective method so the actual code for this ticket is very small.

Because both `GetPublicChannelsForTeam()` and `GetTeamMembers()` didn't had unit tests until now, I went ahead and added them. I _could_ add tests to `app/plugin_api_test.go` but these tests would mostly be a copy of the tests I added to `team_test.go` and `channel_test.go`.

#### Open Questions
I would like replace `*model.ChannelList` in the return value of `GetPublicChannelsForTeam()` with `[]*model.Channel`. This would make it easier to plugin developer to work with the method as they don't have to figure out what exactly `model.ChannelList` is. ~~Note that this would hard break old plugins. Which might be a good thing.~~ I think, it would not break it.

#### Ticket Link
Fixes #9814

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jwilander** [comment]
@amyblais we'll need to make a note of this in the changelog for 5.6

File: app/team_test.go [hunk 1]
```diff
@@ -683,3 +687,44 @@ func TestAppUpdateTeamScheme(t *testing.T) {
...
+		// Store the user ids for comparison later
+		userIDs = append(userIDs, ruser.Id)
+	}
+	// Sort them because the result of GetTeamMembers() is also sorted
+	sort.Sort(userIDs)
+
+	// Fetch team members multipile times
+	members, err := th.App.GetTeamMembers(th.BasicTeam.Id, 0, 5)
+	require.Nil(t, err)
+	members2, err := th.App.GetTeamMembers(th.BasicTeam.Id, 5, 5)
```
Comment (cpanato) [inline app/team_test.go]: this team have 10 members right? can we increase the number of `perPage` to more in the second call? just to make sure we are getting 10 members in total

or make another `GetTeamMembers` call with (10, 5) and should not return any member i guess

Comment (hanzei) [reply app/team_test.go]: Sure, fill fix

**Comment by hanzei** [comment]
@cpanato Thoughts on my Open Question?

**Comment by cpanato** [comment]
ChannelList is `type ChannelList []*Channel` if you change to `[]*model.Channel` is not the same thing?
I might be wrong 😂

**Comment by hanzei** [comment]
Sure, it the same. But plugin developer don't know this. They would have to look this up.

**Comment by cpanato** [comment]
but he will need to lookup anyways what is channel.

i'm 0/5 with this change. nothing against.

**Comment by jwilander** [comment]
+1 for changing to `[]*model.Channel`

**Comment by hanzei** [comment]
Changed it.

**Comment by amyblais** [comment]
@jwilander Is this a breaking change, and important upgrade note, or just needed to be added to the plugins section in the changelog?

**Comment by jwilander** [comment]
@amyblais It's a breaking change in the plugin API, this should work as the note:

```
// in the plugin API section
- Changed `GetTeamMembers(teamId string, offset, limit int)` to `GetTeamMembers(teamId string, page, perPage int)` to be clearer and consistent with other APIs
- Changed `GetPublicChannelsForTeam(teamId string, offset, limit int)` to `GetPublicChannelsForTeam(teamId string, page, perPage int)` to be clearer and more consistent with other APIs
```

**Pull request merged**